src/main/scala/edu/ie3/util/osm/model/OsmContainer.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.util.osm.model
+
+import com.typesafe.scalalogging.LazyLogging
+import edu.ie3.util.exceptions.OsmException
+import edu.ie3.util.osm.model.OsmEntity
+import edu.ie3.util.osm.model.CommonOsmKey.{Building, Highway, Landuse}
+import edu.ie3.util.osm.model.OsmContainer.ParOsmContainer.{
+  ExtendedParOsmContainer,
+  SimpleParOsmContainer
+}
+import edu.ie3.util.osm.model.OsmContainer.SeqOsmContainer.{
+  ExtendedSeqOsmContainer,
+  SimpleSeqOsmContainer
+}
+import edu.ie3.util.osm.model.OsmContainer.{ParOsmContainer, SeqOsmContainer}
+import edu.ie3.util.osm.model.OsmEntity.ComposedEntity.Relation.RelationMember.ExtendedRelationMember
+import edu.ie3.util.osm.model.OsmEntity.ComposedEntity.Relation.{
+  ExtendedRelation,
+  RelationMemberType,
+  SimpleRelation
+}
+import edu.ie3.util.osm.model.OsmEntity.ComposedEntity.Way.ExtendedWay
+import edu.ie3.util.osm.model.OsmEntity.ComposedEntity.{ExtendedWay, SimpleWay}
+import edu.ie3.util.osm.model.OsmEntity.{
+  ComposedEntity,
+  ExtendedOsmEntity,
+  Node
+}
+
+import scala.collection.mutable
+import scala.collection.parallel.CollectionConverters.*
+import scala.collection.parallel.immutable.{ParMap, ParSeq}
+import scala.util.{Failure, Success, Try}
+
+sealed trait OsmContainer {
+
+  def nodeFromId(osmId: Long): Option[Node]
+
+  def nodesFromIds(osmIds: Seq[Long]): Seq[Option[Node]]
+
+}
+
+object OsmContainer {
+
+  sealed trait SeqOsmContainer {
+    val nodes: Seq[Node]
+  }
+
+  sealed trait ParOsmContainer {
+    val nodes: ParSeq[Node]
+  }
+
+  sealed trait SimpleContainer extends OsmContainer {
+
+    def par(): SimpleContainer with ParOsmContainer
+
+    def seq(): SimpleContainer with SeqOsmContainer
+
+  }
+
+  sealed trait ExtendedContainer extends OsmContainer {
+
+    def par(): ExtendedContainer with ParOsmContainer
+
+    def seq(): ExtendedContainer with SeqOsmContainer
+
+  }
+
+  object SeqOsmContainer {
+
+    // todo JH: conversion method to transfer simple to extended container
+
+    final case class SimpleSeqOsmContainer(
+        override val nodes: Seq[Node],
+        ways: Seq[SimpleWay],
+        relations: Seq[SimpleRelation]
+    ) extends SimpleContainer
+        with SeqOsmContainer
+        with LazyLogging
+        with ContainerCache {
+
+      private val _wayCache: mutable.Map[Long, ExtendedWay] =
```

Comment:
What are those caches for? They are currently unused...

Reply:
Old code before the trait - good catch. thx! :)